R/functions.R

```diff
@@ -314,17 +327,17 @@ annotation_label_transfer <- function(input_read_RNA_assay,
 #' @importFrom purrr map
 #' @importFrom magrittr not
 #' @importFrom Matrix colSums
-#' @importFrom magrittr extract2
-#' @importFrom SummarizedExperiment assay
-#' @importFrom SummarizedExperiment colData
+#' @importFrom magrittr extract2 not
+#' @importFrom SummarizedExperiment assay colData assay<-
 #' @importFrom tidyselect all_of
 #' 
 #' @export
 alive_identification <- function(input_read_RNA_assay,
                                  empty_droplets_tbl,
                                  annotation_label_transfer_tbl = NULL,
                                  annotation_column = NULL,
-                                 assay = NULL) {
+                                 assay = NULL,
+                                 gene_nomenclature) {
```

Comment:
`gene_nomenclature` is a target in this case (I don't remember if I told you to put it as a dynamic argument, apologies for that in case).

It is better as a target because it is used many many times within the pipeline.

Reply:
did you mean `gene_nomenclature` need to be tiered, or it needs to be put as a module argument?

Reply:
gene_nomenclature should be a target that is not tiered, is just N=1

Reply:
yeah I have defined it as a target in the main function `initialise_hpc` unless I misunderstand your point.

Reply:
if you defined it in `initialise_hpc` there is no need to define it again

Reply:
If the `gene_nomenclature` parameter is absent in the function, there are certain operations in the function that need minor tweak based on `nomenclature`, like the cc.gene transformation. How does the target module decide which `nomenclature` to use?